### PR TITLE
Feature/305 & Feature/306: All invoices view - action panel with filtering and row selection

### PR DESCRIPTION
**Pull request by jluukka**
Feature/305: Action panel for All invoices containing:
- "View" selection and invoice filtering functionality
- Buttons for taking the user to posting or matching view of selected invoice (not yet functional)

Feature/306: Row selection for All invoices list

### CODE REVIEW

File: src/client/components/InvoiceGrid/InvoiceGrid.react.js [hunk 1]
```diff
@@ -2,62 +2,94 @@ import React, { PropTypes } from 'react';
...
+  viewFilters: state.invoiceGrid.viewList[state.invoiceGrid.selectedView].filters
 });
 
 /**
  * Rendering part of invoice grid
  *
  * @author Daniel Zhitomirsky
  */
-const InvoiceGrid = ({ purchaseInvoices, error, loading }, { i18n }) => {
+// eslint-disable-next-line max-len
```
Comment (ilkkalehtinen) [inline src/client/components/InvoiceGrid/InvoiceGrid.react.js]: I would split long lines. It's easier to review etc.

File: src/client/components/InvoiceGrid/reducer.js [hunk 2]
```diff
@@ -1,13 +1,47 @@
...
 } from './constants';
 
 const initialState = {
   purchaseInvoices: undefined,
   loading: false,
-  error: undefined
+  error: undefined,
+  selectedInvoice: -1,
+  selectedView: "ALL_INVOICES",
+  viewList: {
```
Comment (dzhitomirsky-sc) [inline src/client/components/InvoiceGrid/reducer.js]: view list is not a really good name for a object - i thing 'views' - makes more cense.

File: src/client/components/InvoiceGrid/InvoiceGridActionPanel.react.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  return (
+    <div id="invoice-grid-action-panel">
+      <div className="group left">
+        <div className="item sm text-right">{i18n.getMessage("InvoiceGrid.actionPanel.selectView.label")}</div>
+        <div className="item lg">
+          <Select
+            value={selectedView}
+            searchable={false}
+            clearable={false}
+            options={Object.keys(viewList).map((key) => ({ label: viewList[key].label, value: viewList[key].id }))}
```
Comment (dzhitomirsky-sc) [inline src/client/components/InvoiceGrid/InvoiceGridActionPanel.react.js]: views list labels should be translated.

Comment (dzhitomirsky-sc) [inline src/client/components/InvoiceGrid/InvoiceGridActionPanel.react.js]: Why do you need 'Objectc.keys' to extract values if you call 'Object.values' directrly?

File: src/client/components/InvoiceGrid/reducer.js [hunk 4]
```diff
@@ -1,13 +1,47 @@
...
+    "ALL_INVOICES": {
+      id: "ALL_INVOICES",
+      label: "All invoices",
+      filters: []
+    },
+    "FULLY_MATCHED_INVOICES": {
```
Comment (dzhitomirsky-sc) [inline src/client/components/InvoiceGrid/reducer.js]: Extract view names to some constans file.

**Pull request merged**